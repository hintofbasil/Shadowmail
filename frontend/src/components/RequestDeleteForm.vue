<template>
  <div>
    <div class="row">
      <div class="offset-by-two eight columns">
        <form method="POST" id="click-me-form">
          <input
            type="email"
            name="email"
            id="click-me-text-input"
            placeholder="email@shadowmail.co.uk"
            v-model="email"
          />
          <br />
          <input
            type="submit"
            id="click-me-button"
            value="Send request"
            v-on:click="submit"
          />
        </form>
      </div>
    </div>
    <DeleteFormResponse
      v-if="status"
      v-bind:status="status"
      v-bind:reason="reason"
    />
  </div>
</template>

<script>
import axios from "axios";
import DeleteFormResponse from "@/components/DeleteFormResponse.vue";

const REQUEST_DELETE_URL = "/api/request_delete";

export default {
  components: {
    DeleteFormResponse
  },
  data: () => {
    const url = new URL(document.URL);
    const email = url.searchParams.get("email");
    return {
      email: email,
      status: "",
      reason: ""
    };
  },
  methods: {
    submit: async function(event) {
      event.preventDefault();
      const data = {
        email: this.email
      };
      try {
        const response = await axios.post(REQUEST_DELETE_URL, data);
        this.status = response.data.status;
        this.reason = "Confirmation email sent";
      } catch (error) {
        const { response } = error;
        this.status = response.data.status || "ERROR";
        this.reason = response.data.reason || "An unexpected error occured";
      }
    }
  }
};
</script>

<style lang="scss" scoped>
@import "../colors";

#click-me-button {
  max-width: 90%;
  word-wrap: break-word;
  white-space: normal;
  height: 100%;
  margin-bottom: 0;
}

#click-me-form {
  text-align: center;
  padding-top: 10%;
  padding-bottom: 10%;
  margin-bottom: 0;
}

#click-me-text-input {
  text-align: center;
  margin-bottom: 5rem;
  width: 100%;
  font-size: 1.4em;
}

@media (min-width: 550px) {
  #click-me-form {
    padding-top: 6%;
    padding-bottom: 6%;
  }

  #click-me-button {
    font-size: 1em;
  }
}
</style>
