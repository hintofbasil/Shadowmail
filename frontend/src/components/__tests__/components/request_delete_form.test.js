import React from 'react';
import renderer from 'react-test-renderer';

import RequestDeleteForm from '../../request_delete_form';

describe('render', () => {
  it('renders successfully', () => {
    const tree = renderer
      .create(
        <RequestDeleteForm />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
