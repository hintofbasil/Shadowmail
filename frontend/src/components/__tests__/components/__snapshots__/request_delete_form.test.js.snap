// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders successfully 1`] = `
<div
  className="row"
>
  <div
    className="offset-by-two eight columns"
  >
    <form
      id="click-me-form"
      method="POST"
    >
      <input
        id="click-me-text-input"
        name="email"
        onChange={[Function]}
        placeholder="you@domain.com"
        type="email"
        value={null}
      />
      <br />
      <input
        id="click-me-button"
        onClick={[Function]}
        type="submit"
        value="Send request"
      />
    </form>
  </div>
</div>
`;
